src/ledger/LedgerManager.h

```diff
@@ -156,6 +157,31 @@ class InMemorySorobanState;
 // the actions taken and variables updated by the main thread _after_ apply, and
 // the term "apply" to refer to actions taken and variables updated by the apply
 // thread.
+
+// Consolidated state for ledger commit operations
+struct LedgerCommitState : public NonCopyable
+{
+    // These fields hold the entries that will be commited to the BucketList as
+    // INIT, LIvE, and DEAD entries.
+    std::vector<LedgerEntry> initEntries;
```

Comment:
I think it might be beneficial to wrap these 3 vectors into a separate struct, now that we're refactoring this (also see my comment in the impl concerning passing several vectors vs a single struct). Probably the rest of the fields could belong to a separate struct as well, though that's less clear.